src/core/components/AuthenticateButton/index.js

```diff
@@ -99,16 +103,21 @@ export const mapStateToProps = (
 });
 
 type DispatchMappedProps = {|
-  handleLogOut: HandleLogOutFunc,
+  handleLogOut: HandleLogOutFunction,
 |};
 
+export const createHandleLogOutFunction = (
```

Comment:
Since this is now a shared unit, it needs tests of its own. This is necessary because it allows us to update its implementation without having to rewrite the tests of each component that uses it.